python/cuml/test/test_fil.py

```diff
@@ -542,36 +546,35 @@ def test_lightgbm(tmp_path, num_classes, n_categorical):
         n_rows = 500
         n_informative = 'auto'
 
+    state = np.random.RandomState(43210)
     X, y = simulate_data(n_rows,
                          n_features,
                          num_classes,
                          n_informative=n_informative,
-                         random_state=43210,
+                         random_state=state,
                          classification=True)
+    rng = np.random.default_rng(hash(state))
     if n_categorical > 0:
-        X_fit, X = to_categorical(X, n_categorical)
+        X_fit, X = to_categorical(X, n_categorical, 10, rng)
```

Comment:
10 here is the percentage of invalid categories, but it's not obvious. It's better to use a named parameter for this or introduce a variable with the appropriate name.

Also, it's better to specify it as a fraction (0.1 in this case).